blocks/subscriptions-block/components/usePaywall.js

```diff
@@ -54,14 +53,25 @@ const usePaywall = () => {
 			contentIdentifier &&
 			contentRestriction &&
 			isIdentityInitialized &&
-			!isPaywalled &&
-			!isServerSide()
+			!isPaywalled
 		) {
 			setTimeout(() => runPaywall(), 1000);
 		}
-	}, [apiOrigin, globalContent, Identity, isIdentityInitialized, isPaywalled]);
+	}, [
+		apiOrigin,
+		globalContent,
+		Identity,
+		isIdentityInitialized,
+		isPaywalled,
+		contentIdentifier,
+		contentRestriction,
+		contentType,
+	]);
+
+	
 
 	useEffect(() => {
+		const rules = window?.ArcP?._rules || [];
```

Comment:
@LauraPinilla , could you add `// eslint-disable-next-line no-underscore-dangle` right before this line. I wish eslint was a little more aware and could tell that prop isn't being named that way here.